pkg/controllers/federatedcluster/clusterstatus.go

```diff
@@ -117,6 +115,47 @@ func collectIndividualClusterStatus(
 	return nil
 }
 
+func (c *FederatedClusterController) getResourceCollectorForCluster(clusterName string) (*clustercollector.ClusterCollector, error) {
+	var collector *clustercollector.ClusterCollector
+
+	if cached, ok := c.resourceCollectors.Load(clusterName); ok {
+		collector = cached.(*clustercollector.ClusterCollector)
+	} else {
+		clusterKubeClient, exists, err := c.federatedClient.KubeClientsetForCluster(clusterName)
+		if !exists {
+			return nil, fmt.Errorf("federated client not up to date")
+		}
+		if err != nil {
+			return nil, fmt.Errorf("failed to get federated kube client: %w", err)
+		}
+
+		clusterKubeInformers, exists, err := c.federatedClient.KubeSharedInformerFactoryForCluster(clusterName)
+		if !exists {
+			return nil, fmt.Errorf("federated client not up to date")
+		}
+		if err != nil {
+			return nil, fmt.Errorf("failed to get federated kube informer factory: %w", err)
+		}
+
+		collector = clustercollector.NewClusterCollector(clusterKubeClient, clusterKubeInformers.Core().V1().Nodes())
+		collector.Start(context.TODO())
+		c.resourceCollectors.Store(clusterName, collector)
+	}
+
+	return collector, nil
+}
+
+func (c *FederatedClusterController) deleteResourceCollectorForCluster(clusterName string) {
+	cached, ok := c.resourceCollectors.Load(clusterName)
+	if !ok {
+		return
+	}
+
+	collector := cached.(*clustercollector.ClusterCollector)
+	collector.Stop()
+	c.resourceCollectors.Delete(clusterName)
```

Comment:
Do we need to `LoadAndDelete` here? Any chance for a race condition between `getResourceCollectorForCluster` and `deleteResourceCollectorForCluster`?

Reply:
Both methods are called from 2 different workers.

Reply:
Hmmm, I don't see a race condition. Could you be more specific?

Reply:
> Both methods are called from 2 different workers.

I think I might have misread the code.

Will these 2 methods ever be called for the same cluster on 2 goroutines? If not, there won't be a race condition.